feat: zksync compatibility checks [ZK-004]

ljankovic-txfusion opened the pull request:
### Description

This PR enhances ISM deployments and transaction estimation with:
- Technical stack compatibility verification for ISM types
- Static address support in ISM factory deployments 
- Gas estimation skip for zkSync deployments


### Drive-by changes

- Added `isIsmCompatible` utility function import
- Added technical stack validation checks

### Related issues

N/A

### Backward compatibility

Yes

### Testing

Build passing

mortezashojaei committed:
fix: skip gas estimation on zkSync due to signer requirement

mortezashojaei committed:
feat: add technical stack-aware ISM factory deployment with static address support

mortezashojaei committed:
feat: Add technical stack compatibility check for ISM deployment

mortezashojaei committed:
docs(changeset): Check for ZKSync contracts and functionalities support

ljankovic-txfusion committed:
Merge remote-tracking branch 'origin/feat/zksync-verification-classes' into feat/zksync-compatibility-check

ljankovic-txfusion committed:
Merge remote-tracking branch 'origin/main' into feat/zksync-compatibility-check

ljankovic-txfusion committed:
Merge remote-tracking branch 'origin/main' into feat/zksync-compatibility-check

ljankovic-txfusion committed:
Merge remote-tracking branch 'origin/feat/zksync-verification-classes' into feat/zksync-compatibility-check

ljankovic-txfusion committed:
Merge remote-tracking branch 'origin/feat/zksync-verification-classes' into feat/zksync-compatibility-check

ljankovic-txfusion committed:
Merge remote-tracking branch 'origin/feat/zksync-verification-classes' into feat/zksync-compatibility-check

ljankovic-txfusion committed:
Merge remote-tracking branch 'origin/feat/zksync-verification-classes' into feat/zksync-compatibility-check

ljankovic-txfusion committed:
Merge remote-tracking branch 'origin/feat/zksync-verification-classes' into feat/zksync-compatibility-check

ljankovic-txfusion committed:
Merge remote-tracking branch 'origin/feat/zksync-verification-classes' into feat/zksync-compatibility-check

ljankovic-txfusion committed:
Merge remote-tracking branch 'origin/feat/zksync-verification-classes' into feat/zksync-compatibility-check

ljankovic-txfusion committed:
Merge remote-tracking branch 'origin/feat/zksync-verification-classes' into feat/zksync-compatibility-check

ljankovic-txfusion committed:
Merge remote-tracking branch 'origin/feat/zksync-verification-classes' into feat/zksync-compatibility-check

ljankovic-txfusion committed:
Merge remote-tracking branch 'origin/feat/zksync-verification-classes' into feat/zksync-compatibility-check

ljankovic-txfusion committed:
Merge remote-tracking branch 'origin/feat/zksync-verification-classes' into feat/zksync-compatibility-check

ljankovic-txfusion committed:
chore: update Solidity version in zk-hardhat.config.cts to 0.8.22

ljankovic-txfusion committed:
Merge remote-tracking branch 'origin/feat/zksync-verification-classes' into feat/zksync-compatibility-check

ljankovic-txfusion committed:
Merge remote-tracking branch 'origin/feat/zksync-verification-classes' into feat/zksync-compatibility-check

ljankovic-txfusion committed:
Merge remote-tracking branch 'origin/feat/zksync-verification-classes' into feat/zksync-compatibility-check

ljankovic-txfusion committed:
Merge remote-tracking branch 'origin/feat/zksync-verification-classes' into feat/zksync-compatibility-check

ljankovic-txfusion committed:
Merge remote-tracking branch 'origin/feat/zksync-verification-classes' into feat/zksync-compatibility-check

ljankovic-txfusion committed:
Merge remote-tracking branch 'origin/feat/zksync-verification-classes' into feat/zksync-compatibility-check

ljankovic-txfusion committed:
Merge remote-tracking branch 'origin/feat/zksync-verification-classes' into feat/zksync-compatibility-check

ljankovic-txfusion committed:
Merge remote-tracking branch 'origin/feat/zksync-verification-classes' into feat/zksync-compatibility-check

ljankovic-txfusion committed:
Merge remote-tracking branch 'origin/feat/zksync-verification-classes' into feat/zksync-compatibility-check

ljankovic-txfusion committed:
Merge remote-tracking branch 'origin/feat/zksync-verification-classes' into feat/zksync-compatibility-check

ljankovic-txfusion committed:
Merge remote-tracking branch 'origin/feat/zksync-verification-classes' into feat/zksync-compatibility-check

ljankovic-txfusion committed:
Merge remote-tracking branch 'origin/feat/zksync-verification-classes' into feat/zksync-compatibility-check

ljankovic-txfusion committed:
Merge remote-tracking branch 'origin/feat/zksync-verification-classes' into feat/zksync-compatibility-check

paulbalaji commented on typescript/sdk/src/ism/HyperlaneIsmFactory.ts:
```diff
@@ -149,6 +149,14 @@ export class HyperlaneIsmFactory extends HyperlaneApp<ProxyFactoryFactories> {
+
+    // For static ISM types it checks whether the technical stack supports static contract deployment
+    assert(
+      isIsmCompatible({ chainTechnicalStack: technicalStack, ismType }),
```
should we be checking if the hooks are compatible too?

ljankovic-txfusion replied:
I have not run into ever setting static hooks on core-config or warp-config, usually check for static ism is enough. How would we go about deploying static hooks from cli perspective?

paulbalaji replied:
I guess it's just a check that we're not using storageaggregation vs staticaggregation hooks, i'm not sure if there are are other static hooks we care about

ljankovic-txfusion replied:
Ah now I understand why I did not catch it the first time, for ISM it says staticAggregationIsm and for hook it just says aggregationHook. You are right! I will include hook type validation

ljankovic-txfusion committed:
Merge remote-tracking branch 'origin/feat/zksync-verification-classes' into feat/zksync-compatibility-check

ljankovic-txfusion committed:
Merge remote-tracking branch 'origin/feat/zksync-verification-classes' into feat/zksync-compatibility-check

ljankovic-txfusion committed:
refactor: reorganize ISM utility exports and add isStaticIsm function

ljankovic-txfusion committed:
feat: add isHookCompatible function to check hook type compatibility with chain's technical stack

ljankovic-txfusion committed:
refactor: move isHookCompatible function to a new file and update ISM utility exports

ljankovic-txfusion committed:
Merge remote-tracking branch 'origin/feat/zksync-verification-classes' into feat/zksync-compatibility-check

ljankovic-txfusion committed:
Merge remote-tracking branch 'origin/feat/zksync-verification-classes' into feat/zksync-compatibility-check

ljankovic-txfusion committed:
Merge remote-tracking branch 'origin/feat/zksync-verification-classes' into feat/zksync-compatibility-check

ljankovic-txfusion committed:
Merge remote-tracking branch 'origin/feat/zksync-verification-classes' into feat/zksync-compatibility-check

ljankovic-txfusion committed:
Merge remote-tracking branch 'origin/feat/zksync-verification-classes' into feat/zksync-compatibility-check

daniel-savu committed:
chore: evm e2e with new submitter (#6196)

### Description

<!--
What's included in this PR?
-->

### Drive-by changes

<!--
Are there any minor or drive-by changes also included?
-->

### Related issues

- Fixes https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/6147

### Backward compatibility

<!--
Are these changes backward compatible? Are there any infrastructure
implications, e.g. changes that would prohibit deploying older commits
using this infra tooling?

Yes/No
-->

### Testing

<!--
What kind of testing have these changes undergone?

None/Manual/Unit Tests
-->

---------

Co-authored-by: Danil Nemirovsky <4614623+ameten@users.noreply.github.com>

paulbalaji committed:
fix: storage diff-check (#6290)

### Description

fix: storage diff-check

### Drive-by changes

<!--
Are there any minor or drive-by changes also included?
-->

### Related issues

<!--
- Fixes #[issue number here]
-->

### Backward compatibility

<!--
Are these changes backward compatible? Are there any infrastructure
implications, e.g. changes that would prohibit deploying older commits
using this infra tooling?

Yes/No
-->

### Testing

<!--
What kind of testing have these changes undergone?

None/Manual/Unit Tests
-->

paulbalaji committed:
chore: depot docker recommendations (#6258)

### Description

chore: depot docker recommendations
- update dockerfiles based on the depot recommendations

### Drive-by changes

<!--
Are there any minor or drive-by changes also included?
-->

### Related issues

<!--
- Fixes #[issue number here]
-->

### Backward compatibility

<!--
Are these changes backward compatible? Are there any infrastructure
implications, e.g. changes that would prohibit deploying older commits
using this infra tooling?

Yes/No
-->

### Testing

- [x] ci
- [x] monorepo image on keyfunder
- [x] agent image on rc

paulbalaji committed:
feat: setup turborepo (#3899)

- use [turbo](https://turbo.build/repo/docs) for faster build times
	- automatic build task scheduling/dependency ordering
	- automatic build caching
	- faster build times
	- see turbo docs for more info
	- better integration with depot cache
- use depot's turbo cache https://depot.dev/docs/cache/reference/turbo
- use more depot runners to speed up recovery of yarn-build-with-cache
action
- depot runners will make use of depot's caching backend over
github-native or buildjet

troykessler committed:
chore: add cosmos native core cli commands (#6263)

### Description

This PR finally adds support for Cosmos Native Core Init, Read, Deploy &
Apply.

Note that this is the fifth and last PR of
https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/6050 which is
being split up in order to make it easier to review

### Drive-by changes

-

### Related issues

-

### Backward compatibility

Yes

### Testing

Manual Tests

daniel-savu committed:
feat(relayer): optimize ccip-read initialization (#6299)

### Description

https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/6144 made
relayer initialization very slow. It creates a signing provider within
nested for loops, spending quadratic time (based on chain count) to
create message contexts. At 110 chains and 200ms per network call that's
~40 mins (`110*110*200/(1000*60)`).

since the origin and destination chains are identical in the relayer,
the first commit optimizes this by only initializing the provider in the
upper for-loop, making it linear. This got us down to ~30s of
initialization overhead added by the ccip read initialization.

the second commit moves out the provider building out of the for loops
and just awaits the futures in parallel. AWS KMS rate limits shouldn't
be a concern since it's just `chain_count` calls. After releasing to RC,
the initialization overhead dropped to near zero

see the three logs
[here](https://cloudlogging.app.goo.gl/H7LmsVr5hgzJSSdk7) for
performance comparison:
- first log is without the CCIP-read feature (30 sec baseline)
- second log is with linear provider building (60 sec, so 30s overhead)
- third log is with the parallelized provider building (30 sec, back to
the baseline)


### Drive-by changes

<!--
Are there any minor or drive-by changes also included?
-->

### Related issues

<!--
- Fixes #[issue number here]
-->

### Backward compatibility

<!--
Are these changes backward compatible? Are there any infrastructure
implications, e.g. changes that would prohibit deploying older commits
using this infra tooling?

Yes/No
-->

### Testing

<!--
What kind of testing have these changes undergone?

None/Manual/Unit Tests
-->

Mo-Hussain committed:
fix: require ssl for connection (#6285)

### Description

require ssl for connection

### Testing

Manual

daniel-savu committed:
fix: priority gas price overpayments (#6295)

### Description

Applies changes from https://github.com/hyperlane-xyz/ethers-rs/pull/35
- tested on Base and we now pay 80x less

Mo-Hussain committed:
chore: update monitor image to include new token standard handling (#6242)

### Description

- update monitor image to include new token standard handling

christopherbrumm committed:
fix: token metadata handling (#6175)

### Description

This PR fixes the token metadata handling by changing the return type of
`deriveTokenMetadata` to TokenMetadataMap`, enabling support for
multiple different token metadata.

### Drive-by changes

- `assertWarpRouteConfig` expects the mailbox address to verify the
correct deployed mailbox.
- remove `./test-configs/anvil/deployments` before running e2e-tests


### Backward compatibility

Yes

### Testing

Added E2E test to verify that for `collateralFiat` and `collateral` with
two different metadata, token is deployed with two different metadata.

---------

Co-authored-by: Morteza Shojaei <morteza@txfusion.io>

ameten committed:
fix: Scraper: Fix issue with missing enum invariant for Sealevel (#6298)

### Description

Scraper is failing to index some message dispatches due to obsolete
Solana SDK we are using. We pinned the dependency version to 1.14.13.
Since then enum `TransactionError` added a few enum invariants. New tag
of the Solana SDK dependencies brings these invariants from upstream.

### Backward compatibility

Yes

### Testing

Tested with experimental image

Co-authored-by: Danil Nemirovsky <4614623+ameten@users.noreply.github.com>

daniel-savu committed:
chore: new relayer image (#6304)

### Description

<!--
What's included in this PR?
-->

### Drive-by changes

<!--
Are there any minor or drive-by changes also included?
-->

### Related issues

<!--
- Fixes #[issue number here]
-->

### Backward compatibility

<!--
Are these changes backward compatible? Are there any infrastructure
implications, e.g. changes that would prohibit deploying older commits
using this infra tooling?

Yes/No
-->

### Testing

<!--
What kind of testing have these changes undergone?

None/Manual/Unit Tests
-->

paulbalaji committed:
chore: reduce release-e2e-matrix frequency (#6296)

### Description

chore: reduce release-e2e-matrix frequency
- previously we would run the release cli test matrix if there were no
changesets
- no changesets _implied_ that a release is to be done, but in cases
where packages are published but no new publishable changes have landed
yet - we're still churning through these tests for no good reason
- this PR changes the behaviour such that we always try and do
prepare-release so that the Changesets PR is updated, but we also in
parallel check if the _current_ package versions are published and only
run the tests if we detect that there are packages that need to be
published

### Drive-by changes

<!--
Are there any minor or drive-by changes also included?
-->

### Related issues

<!--
- Fixes #[issue number here]
-->

### Backward compatibility

<!--
Are these changes backward compatible? Are there any infrastructure
implications, e.g. changes that would prohibit deploying older commits
using this infra tooling?

Yes/No
-->

### Testing

tested here
https://github.com/hyperlane-xyz/hyperlane-monorepo/actions/runs/15161126534

![image](https://github.com/user-attachments/assets/984e16d7-ae0e-4cb6-ad0f-2a5adbfc5fd9)

![image](https://github.com/user-attachments/assets/61f960fe-3ee1-4250-b737-0c515c23bb73)

ameten committed:
fix(submitter): Fix transaction fields (#6306)

### Description

Required fields on EVM transaction are overwritten when we estimate gas
price in EVM adapter for Lander. It means that these fields are not set
when transaction is submitted into a chain.

This change ensures that all the required fields are copied from old
transaction to the new one during gas price estimations.

E2E tests do not pass with fields set, so, we use classical submitter
for the time being.

### Backward compatibility

Yes

### Testing

E2E test

Co-authored-by: Danil Nemirovsky <4614623+ameten@users.noreply.github.com>

ameten committed:
chore: Upgrade Scraper to latest (#6307)

### Description

Upgrade Scraper to 197fd87-20250521-144619.

### Backward compatibility

Yes

### Testing

Tested in hyperlane

Co-authored-by: Danil Nemirovsky <4614623+ameten@users.noreply.github.com>

christopherbrumm committed:
chore: add `sol`, `ufd`, `fartcoin`& `pengu` (#6249)

### Description

Add `sol`, `ufd`, `fartcoin` & `pengu` between Solana and Apechain.

### Backward compatibility

Yes

### Testing

N/A

---------

Co-authored-by: xeno097 <xeno097.cp@gmail.com>

mshojaei-txfusion committed:
Merge branch 'main' into feat/zksync-verification-classes

mshojaei-txfusion committed:
feat: yaml sorting eslint plugin (#5909)

### Description

This PR adds a new ESLint rule `sort-yaml-arrays` to the utils package,
which allows for sorting arrays in YAML files based on specified keys.
This helps maintain consistent and predictable ordering in YAML
configuration files.

Key features:
- New utility function for array sorting by key
- YAML transformation with complete comment preservation 
- Support for sorting nested arrays at specific paths
- Comprehensive test coverage for all sorting scenarios and comment
handling

### Drive-by changes

- Enhanced the comment preservation logic in YAML transformation
- Improved unit test coverage for YAML transformation utilities

### Related issues
Required to resolve
https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/5547

### Backward compatibility

Yes - This change is fully backward compatible. It adds a new ESLint
rule that can be optionally enabled.

### Testing

Unit Tests - Comprehensive test suite covers:
- Basic array sorting functionality
- Nested array sorting
- Comment preservation
- Various edge cases

troykessler committed:
fix: disable cosmos signers for now (#6309)

### Description

This PR only inits Ethereum signers and temporary reverts the changes
made in https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/6124.
This is necessary, since Cosmos Signers are being initialized with an
Eth private key in some circumstances which throw an error. The cosmos
signers should therefore be disabled until a solution has been found to
manage multi-vm signer intializations.

### Drive-by changes

-

### Related issues

-

### Backward compatibility

Yes

### Testing

Manual Tests

---------

Co-authored-by: Paul Balaji <10051819+paulbalaji@users.noreply.github.com>

troykessler committed:
chore: add routing ism to cosmos native ism module (#6264)

### Description

This PR add the Cosmos Native Routing ISM to the CosmosNativeIsmModule.

### Drive-by changes

-

### Related issues

-

### Backward compatibility

Yes

### Testing

Manual Tests

---------

Co-authored-by: graphite-app[bot] <96075541+graphite-app[bot]@users.noreply.github.com>

paulbalaji committed:
chore: change extractPrivateKey info logs to debug logs (#6310)

### Description

chore: reduce logging of extractPrivateKey

### Drive-by changes

<!--
Are there any minor or drive-by changes also included?
-->

### Related issues

<!--
- Fixes #[issue number here]
-->

### Backward compatibility

<!--
Are these changes backward compatible? Are there any infrastructure
implications, e.g. changes that would prohibit deploying older commits
using this infra tooling?

Yes/No
-->

### Testing

<!--
What kind of testing have these changes undergone?

None/Manual/Unit Tests
-->

antigremlin committed:
chore: migrate to workspace dependency syntax (#6308)

### Description

This PR migrates all internal monorepo package dependencies from
explicit version numbers (e.g., `"@hyperlane-xyz/sdk": "10.0.0"`) to the
recommended `workspace:^` syntax (e.g., `"@hyperlane-xyz/sdk":
"workspace:^"`).

Details:

- When working locally, Yarn will always symlink to the latest version
of the dependency within the monorepo, regardless of its declared
version. This eliminates the need for manual version bumps in
package.json files during active development (although we managed these
version bumps with `changeset version`).
- When packages are published, Yarn automatically transforms
`workspace:^` into a standard semver range (e.g., `^1.2.3`).

### Related issues

Fixes ENG-1666

### Backward compatibility

- `yarn changeset version` continues to work as before.

### Testing

- Manually tested code navigation in VSCode
- Manually tested `yarn changeset version`

paulbalaji committed:
chore: disable arthera (#6313)

### Description

chore: disable arthera

### Drive-by changes

<!--
Are there any minor or drive-by changes also included?
-->

### Related issues

<!--
- Fixes #[issue number here]
-->

### Backward compatibility

<!--
Are these changes backward compatible? Are there any infrastructure
implications, e.g. changes that would prohibit deploying older commits
using this infra tooling?

Yes/No
-->

### Testing

<!--
What kind of testing have these changes undergone?

None/Manual/Unit Tests
-->

mshojaei-txfusion committed:
Merge remote-tracking branch 'origin/feat/zksync-verification-classes' into feat/zksync-compatibility-check

antigremlin committed:
chore: share tsconfig.json from a package (#6314)

### Description

This PR moves the shared TypeScript configuration into a NPM private
package inside the monorepo, as recommended by Turborepo. The package
isn't going to be published, just reused internally. This removes the
need to reference parent paths in all our TypeScript config files.

### Related issues

Part of: ENG-1668

### Backward compatibility

Yes, as the TypeScript configuration stays unchanged.

### Testing

All TypeScript builds continue to work.

ameten committed:
fix: Make Solana testnet chunk same as for mainnet (#6311)

### Description

Make Solana testnet chunk same as for mainnet

### Backward compatibility

Yes

### Testing

Manual

Co-authored-by: Danil Nemirovsky <4614623+ameten@users.noreply.github.com>

paulbalaji committed:
feat: may 18 multisig batch (#6259)

### Description

feat: may 18 multisig batch
- migrate nibiru ownership
- enroll new chains hashkey, infinityvmmainnet, game7, fluence, peaq,
ontology, miraclechain, kyve, svmbnb

### Drive-by changes

- improved retry+logging when fetching txs in infra safe utils
- required because safe apis are so useless that the parsing script
would repeatedly and instantly fail without adding this additional retry
logic
- retry logic originally used when reviewing the ousdt extension txs,
and is part of a broader set of updates found here
https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/6020

### Related issues

https://github.com/hyperlane-xyz/hyperlane-registry/pull/863

### Backward compatibility

<!--
Are these changes backward compatible? Are there any infrastructure
implications, e.g. changes that would prohibit deploying older commits
using this infra tooling?

Yes/No
-->

### Testing

<!--
What kind of testing have these changes undergone?

None/Manual/Unit Tests
-->

daniel-savu committed:
feat: agent-specific rpc consensus defaults (#6316)

### Description

We recently discovered that validators use fallback provider as the
default consensus type, which leaves them susceptible to reorgs
regardless of how big their RPC set is.

This PR adds agent-specific rpc consensus defaults: fallback for the
relayer and scraper, quorum for the validator. This default can still be
overriden via the regular config.

I expect this will increase checkpoint-posting latency for validators
that have some bad RPCs in their set.

### Drive-by changes

The `defaultRpcConsensusType` is removed from the agent config schema,
since it applied to all chains and agents, which is much too generic at
this point. It was always set to `fallback` in our infra TS code. It now
still defaults to `fallback` in all cases but when the validator is run.

### Backward compatibility

The agent schema no longer contains `defaultRpcConsensusType`. See
above. This won't break any builds with new agents that use old configs.

### Testing

Manual, by running a local validator and printing the
`origin_chain_conf` to make sure it's using `HttpQuorum`

paulbalaji committed:
feat: may 21 testnet batch (#6300)

### Description

feat: may 21 testnet batch
- add rometestnet, neuratestnet
- remove suavetoliman

### Drive-by changes

- rometestnet is only connected to 5 chains so required manually
tweaking of testnet igp/core config generation for rometestnet
- enable scraping of solana testnet

### Related issues

https://github.com/hyperlane-xyz/hyperlane-registry/pull/874

### Backward compatibility

<!--
Are these changes backward compatible? Are there any infrastructure
implications, e.g. changes that would prohibit deploying older commits
using this infra tooling?

Yes/No
-->

### Testing

<!--
What kind of testing have these changes undergone?

None/Manual/Unit Tests
-->

paulbalaji committed:
feat: may 20 svm multisig batch (#6283)

### Description

feat: SVM ism/igp updates
- add subtensor to solana igp
- update solana ism
- update soon ism
- update svmbnb ism

### Drive-by changes

- when deploying isms, get local_domain from chain-config-file instead
of KnownHyperlaneDomains

### Related issues

https://github.com/hyperlane-xyz/hyperlane-registry/pull/871

### Backward compatibility

<!--
Are these changes backward compatible? Are there any infrastructure
implications, e.g. changes that would prohibit deploying older commits
using this infra tooling?

Yes/No
-->

### Testing

<!--
What kind of testing have these changes undergone?

None/Manual/Unit Tests
-->

paulbalaji committed:
chore: trigger release PR testing automatically (#6318)

### Description

chore: trigger release PR testing automatically

Currently due to the nature of the changesets bot, it requires
closing/reopening the release PR to trigger the tests for CI to run.
Hoping that enabling this will cause CI to automatically run on the
release PR branch

### Drive-by changes

<!--
Are there any minor or drive-by changes also included?
-->

### Related issues

<!--
- Fixes #[issue number here]
-->

### Backward compatibility

<!--
Are these changes backward compatible? Are there any infrastructure
implications, e.g. changes that would prohibit deploying older commits
using this infra tooling?

Yes/No
-->

### Testing

<!--
What kind of testing have these changes undergone?

None/Manual/Unit Tests
-->

github-actions[bot] committed:
Version Packages (#6302)

This PR was opened by the [Changesets
release](https://github.com/changesets/action) GitHub action. When
you're ready to do a release, you can merge this and publish to npm
yourself or [setup this action to publish
automatically](https://github.com/changesets/action#with-publishing). If
you're not ready to do a release yet, that's fine, whenever you add more
changesets to main, this PR will be updated.


# Releases
## @hyperlane-xyz/cli@13.1.0

### Minor Changes

-   ac13dde: add cosmos native core cli commands

### Patch Changes

- 754f7ba: disable cosmos native signers and only allow evm signers for
now again
- 86a8502: Change info logs to debug logs in
MultiProtocolSignerManager.extractPrivateKey().

## @hyperlane-xyz/sdk@13.1.0

### Minor Changes

- 6e86efa: Remove `defaultRpcConsensusType` from the agent config schema
-   c42ea09: Deploy to new chains: neuratestnet, rometestnet.

### Patch Changes

-   Updated dependencies [f41f766]
    -   @hyperlane-xyz/utils@13.1.0
    -   @hyperlane-xyz/core@7.1.7
    -   @hyperlane-xyz/starknet-core@13.1.0
    -   @hyperlane-xyz/cosmos-sdk@13.1.0

## @hyperlane-xyz/utils@13.1.0

### Minor Changes

-   f41f766: Add sort-yaml-arrays eslint rule

## @hyperlane-xyz/core@7.1.7

### Patch Changes

-   Updated dependencies [f41f766]
    -   @hyperlane-xyz/utils@13.1.0

## @hyperlane-xyz/cosmos-sdk@13.1.0

### Patch Changes

-   @hyperlane-xyz/cosmos-types@13.1.0

## @hyperlane-xyz/helloworld@13.1.0

### Patch Changes

-   Updated dependencies [6e86efa]
-   Updated dependencies [c42ea09]
    -   @hyperlane-xyz/sdk@13.1.0
    -   @hyperlane-xyz/core@7.1.7

## @hyperlane-xyz/widgets@13.1.0

### Patch Changes

-   Updated dependencies [6e86efa]
-   Updated dependencies [c42ea09]
-   Updated dependencies [f41f766]
    -   @hyperlane-xyz/sdk@13.1.0
    -   @hyperlane-xyz/utils@13.1.0
    -   @hyperlane-xyz/cosmos-sdk@13.1.0

## @hyperlane-xyz/starknet-core@13.1.0



## @hyperlane-xyz/cosmos-types@13.1.0



## @hyperlane-xyz/tsconfig@13.1.0

### Minor Changes

- 7d5c6b4: Share TypeScript configuration has moved into a private
package

## @hyperlane-xyz/infra@13.1.0

### Patch Changes

-   Updated dependencies [6e86efa]
-   Updated dependencies [c42ea09]
-   Updated dependencies [f41f766]
    -   @hyperlane-xyz/sdk@13.1.0
    -   @hyperlane-xyz/utils@13.1.0
    -   @hyperlane-xyz/helloworld@13.1.0

## @hyperlane-xyz/ccip-server@13.1.0



## @hyperlane-xyz/github-proxy@13.1.0

---------

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: pbio <10051819+paulbalaji@users.noreply.github.com>

paulbalaji committed:
feat: publish with github-hosted runner (#6319)

ltyu committed:
feat: update warp init to generate config with new warp route id (#6230)

### Description
This PR enables `warp init` to check that a warpRouteId is valid by
proposing a short or long ID and allow user to create a custom ID. The
main logic is inside of `getWarpRouteIdFromWarpDeployConfig()`.

Relies on: https://github.com/hyperlane-xyz/hyperlane-registry/pull/853

### Backward compatibility
Yes - Falls back to current warp route id generation strategy

### Testing
Manual

---------

Co-authored-by: graphite-app[bot] <96075541+graphite-app[bot]@users.noreply.github.com>

paulbalaji committed:
Revert "chore: migrate to workspace dependency syntax (#6308)" (#6326)

This reverts commit 72db0fce57aa51e2faf921937b9cd5e2f69358e2.

### Description

<!--
What's included in this PR?
-->

### Drive-by changes

<!--
Are there any minor or drive-by changes also included?
-->

### Related issues

<!--
- Fixes #[issue number here]
-->

### Backward compatibility

<!--
Are these changes backward compatible? Are there any infrastructure
implications, e.g. changes that would prohibit deploying older commits
using this infra tooling?

Yes/No
-->

### Testing

<!--
What kind of testing have these changes undergone?

None/Manual/Unit Tests
-->

github-actions[bot] committed:
Version Packages (#6327)

mshojaei-txfusion committed:
Merge branch 'main' into feat/zksync-compatibility-check
